src/Java.Runtime.Environment/Java.Interop/JreRuntime.cs

```diff
@@ -99,11 +97,21 @@ static unsafe JreRuntimeOptions CreateJreVM (JreRuntimeOptions builder)
 				builder.ObjectReferenceManager  = builder.ObjectReferenceManager    ?? new ManagedObjectReferenceManager (builder.JniGlobalReferenceLogWriter, builder.JniLocalReferenceLogWriter);
 			}
 
-			if (builder.InvocationPointer != IntPtr.Zero)
+			if (builder.InvocationPointer != IntPtr.Zero || builder.EnvironmentPointer != IntPtr.Zero)
 				return builder;
 
 			builder.LibraryHandler.LoadJvmLibrary (builder.JvmLibraryPath!);
 
+			if (!builder.ClassPath.Any (p => p.EndsWith ("java-interop.jar", StringComparison.OrdinalIgnoreCase))) {
+				var loc = typeof (JreRuntimeOptions).Assembly.Location;
+				var dir = string.IsNullOrEmpty (loc) ? null : Path.GetDirectoryName (loc);
+				var jij = string.IsNullOrEmpty (dir) ? null : Path.Combine (dir, "java-interop.jar");
+				if (!File.Exists (jij)) {
+					throw new InvalidOperationException ($"`java-interop.jar` is required.  Please add to `JreRuntimeOptions.ClassPath`.  Tried to find it in `{jij}`.");
```

Comment:
Is it worth doing `FileNotFoundException`?

Reply:
Updated.